ingestion_server/ingestion_server/indexer.py

```diff
@@ -169,64 +171,46 @@ def get_last_item_ids(table):
+        task_id: Optional[str] = None,
+        callback_url: Optional[str] = None,
+        progress: Optional[Value] = None,
+        active_workers: Optional[Value] = None,
     ):
         self.es = es_instance
         connections.connections.add_connection("default", self.es)
-        self.tables_to_watch = tables
-        # Optional multiprocessing.Values for examining indexing progress
+
         self.task_id = task_id
+        self.callback_url = callback_url
         self.progress = progress
-        self.finish_time = finish_time
         self.active_workers = active_workers
 
-    def _index_table(self, table, dest_idx=None):
+    # Helpers
+    # =======
+
+    @staticmethod
+    def pg_chunk_to_es(pg_chunk, columns, model_name, dest_index):
         """
-        Check that the database tables are in sync with Elasticsearch. If not,
-        begin replication.
+        Given a list of psycopg2 results, convert them all to Elasticsearch
+        documents.
         """
-        last_pg_id, _ = get_last_item_ids(table)
-        if not last_pg_id:
-            log.warning(f"Tried to sync {table} but it was empty.")
-            return
-        # Find the last document inserted into elasticsearch
-        destination = dest_idx if dest_idx else table
-        s = Search(using=self.es, index=destination)
-        s.aggs.bucket("highest_pg_id", "max", field="id")
-        try:
-            es_res = s.execute()
-            last_es_id = int(es_res.aggregations["highest_pg_id"]["value"])
-        except (TypeError, NotFoundError):
-            log.info(
-                "No matching documents found in elasticsearch. "
-                "Replicating everything."
-            )
-            last_es_id = 0
-        log.info(f"highest_db_id, highest_es_id: {last_pg_id}, {last_es_id}")
-        # Select all documents in-between and replicate to Elasticsearch.
-        if last_pg_id > last_es_id:
-            log.info(f"Replicating range {last_es_id}-{last_pg_id}")
-
-            deleted, mature = get_existence_queries(table)
-            query = SQL(
-                "SELECT *, {deleted}, {mature} "
-                "FROM {table} "
-                "WHERE id BETWEEN {last_es_id} AND {last_pg_id} "
-                "AND license_version IS NOT NULL;"
-            ).format(
-                deleted=deleted,
-                mature=mature,
-                table=Identifier(table),
-                last_es_id=Literal(last_es_id),
-                last_pg_id=Literal(last_pg_id),
-            )
-            self.es.indices.create(index=dest_idx, body=index_settings(table))
-            self.replicate(table, dest_idx, query)
+        # Map column names to locations in the row tuple
+        schema = {col[0]: idx for idx, col in enumerate(columns)}
+        model = database_table_to_elasticsearch_model.get(model_name)
+        if model is None:
+            log.error(f"Table {model_name} is not defined in elasticsearch_models.")
```

Comment:
I'm slightly confused by table/model variable names in this method. It looks like `database_table_to_elasticsearch_model` expects a `table_name` rather than a `model_name`. Should this method take `table_name` instead and pass that here?

Reply:
The `database_table_to_elasticsearch_model` identifier is badly named, it actually takes the name of a media type and returns the ES models for that media. It uses the model name. Updated the name in 9e70ce0.